presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -90,54 +91,65 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
         requireNonNull(symbolAllocator, "symbolAllocator is null");
         requireNonNull(idAllocator, "idAllocator is null");
         if (SystemSessionProperties.isOptimizeHashGenerationEnabled(session)) {
-            PlanWithProperties result = plan.accept(new Rewriter(idAllocator, symbolAllocator, types), new HashSymbolPreference());
+            PlanWithProperties result = plan.accept(new Rewriter(idAllocator, symbolAllocator, types), new HashSymbols());
             return result.getNode();
         }
         return plan;
     }
 
-    private static class HashSymbolPreference
+    private static class HashSymbols
     {
-        private final Set<List<Symbol>> preferredHashSymbols;
+        private final Set<List<Symbol>> hashSymbols;
```

Comment:
so say you have group by on x, then  group by on y, then a group by on z and w, you would have a desire for three hash symbols to be produced, one for x, one for y and one for (z,w)